### DISCUSSION TITLE
use dbpoweramp as transcoder

### DISCUSSION THREAD

**Discussion by blahdy** [navidrome/navidrome#2791] [Q&A]
having trouble setting up dbpoweramp to be used as external transcoder.

i use navidrome for carplay, which is limited to up to 16bit 48khz and the resampling done on the iphone is godawful.

also, downsampling the overhead should help in areas with bad coverage.

but for the life of me I cannot figure out the transcode command for dbpoweramp to be used to downsample

this is what generates the required output in terminal:

/Applications/dBpoweramp\ Music\ Converter.app/Contents/MacOS/dBpoweramp\ Music\ Converter -coreconverter -infile="/Users/user/24-96.m4a" -dspeffect1="Resample=-fmin={qt}44100{qt} -fmax={qt}48000{qt}" -convert_to="FLAC" -dspeffect2="Bit Depth=-mindepth={qt}16{qt} -maxdepth={qt}16{qt} -dither={qt}tpdf{qt}" -outfile="/Users/user/24-96.flac"

i cannot manually get dbpoweramp to print stdout and there is no logging when called from navidrome, just exited with non-zero status code: 1

can somebody pretty please help generate that magic combination of symbols to get it done?

p.s.

the need for dbpoweramp is because it's one of the best resamplers with stdout support, no reason to prefer ffmpeg over the ios

**Comment by deluan** [comment]
You could try to create a shell script similar to this: https://github.com/navidrome/navidrome/issues/2194#issuecomment-1556202775
